homeassistant/components/elevenlabs/tts.py

```diff
@@ -0,0 +1,124 @@
+"""Support for the ElevenLabs text-to-speech service."""
+
+from __future__ import annotations
+
+from functools import cached_property
+import logging
+from typing import Any
+
+from elevenlabs.client import AsyncElevenLabs
+from elevenlabs.core import ApiError
+from elevenlabs.types import Model, Voice
+
+from homeassistant.components import tts
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_MODEL, CONF_VOICE, DEFAULT_MODEL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up ElevenLabs tts platform via config entry."""
+    model_id = config_entry.options.get(CONF_MODEL, config_entry.data.get(CONF_MODEL))
+    default_voice_id = config_entry.options.get(
+        CONF_VOICE, config_entry.data.get(CONF_VOICE)
+    )
+    client = AsyncElevenLabs(api_key=config_entry.data[CONF_API_KEY])
+
+    # Load model and voices here in async context
+    model: Model | None = None
+    models = await client.models.get_all()
+    model_id = model_id if model_id is not None else DEFAULT_MODEL
+    for maybe_model in models:
+        if maybe_model.model_id == model_id:
+            model = maybe_model
+            break
+
+    if (model is None) or (not model.languages):
+        raise ValueError(f"Failed to load model for id: {model_id}")
```

Comment:
You cannot raise errors that the system doesn't know, it will crash things. Test this too inside `async_setup_entry` inside `__init__.py`, so you can `return False` to indicate that the config entry has failed to setup.

Reply:
Implemented.